tests/components/awair/test_config_flow.py

```diff
@@ -82,46 +83,67 @@ async def test_no_devices_error(hass):
         assert result["reason"] == "no_devices_found"
 
 
-async def test_reauth(hass):
+async def test_reauth(hass: HomeAssistant) -> None:
     """Test reauth flow."""
-    with patch(
-        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
-    ), patch(
-        "homeassistant.components.awair.sensor.async_setup_entry",
-        return_value=True,
-    ):
-        mock_config = MockConfigEntry(domain=DOMAIN, unique_id=UNIQUE_ID, data=CONFIG)
-        mock_config.add_to_hass(hass)
-        hass.config_entries.async_update_entry(
-            mock_config, data={**CONFIG, CONF_ACCESS_TOKEN: "blah"}
+    mock_config = MockConfigEntry(
+        domain=DOMAIN, unique_id=UNIQUE_ID, data={**CONFIG, CONF_ACCESS_TOKEN: "blah"}
+    )
+    mock_config.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_REAUTH, "unique_id": UNIQUE_ID},
+        data={**CONFIG, CONF_ACCESS_TOKEN: "blah"},
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "reauth_confirm"
+    assert result["errors"] == {}
+
+    with patch("python_awair.AwairClient.query", side_effect=AuthError()):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=CONFIG,
         )
 
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN,
-            context={"source": SOURCE_REAUTH, "unique_id": UNIQUE_ID},
-            data=CONFIG,
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "reauth_confirm"
+        assert result["errors"] == {CONF_ACCESS_TOKEN: "invalid_access_token"}
+
+    with patch(
+        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
+    ), patch("homeassistant.config_entries.ConfigEntries.async_reload"):
```

Comment:
Let's patch the integration setup function instead. It's less invasive.
```suggestion
    ), patch("homeassistant.components.awair.async_setup_entry", return_value=True):
```